ui/src/components/LogsRow/LogsRow.tsx

```diff
@@ -16,24 +16,43 @@ export default function LogsRow({
   logInfo: logInfo,
   containerLabelColor,
   containerIconColor,
+  ddClient,
 }: {
   logInfo: DockerLog;
   containerLabelColor: Record<string, string>;
   containerIconColor: Record<string, string>;
+  ddClient: DDClient;
 }) {
   const { containerId, containerName, time, stream, log } = logInfo;
   const [open, setOpen] = useState<boolean>(false);
-  const [dummyData, setDummyData] = useState('');
+  const [metrics, setMetrics] = useState('');
 
   const labelColor = containerLabelColor[containerId];
   let iconColor = containerIconColor[containerId] === 'running' ? 'teal' : 'grey';
 
-  const fetchMetrics = () => {
-    // Skeleton for response
-    // const response = await fetch(`/metrics/${containerId}`);
-    const dumbData = 'CPU:3%, RAM:15MB';
-    setDummyData(dumbData);
-    console.log('im working');
+  const fetchMetrics = async (ddClient: DDClient) => {
+    try {
+      // Parse time into Prometheus API friendly format
+      const promTime = (Date.parse(time) / 1000).toFixed(3);
```

Comment:
Question: can you help me understand what's going on here? (Why divide by 1000 and use toFixed)?

Reply:
Prometheus expects certain formatting for its time query parameter.

David will have a more thorough explanation.